usr/sbin/cloudguest-lic-watcher

```diff
@@ -34,29 +35,69 @@ def has_license_changed(license_type):
             return license_type != old_license
     else:
         update_license_cache(license_type)
+        maybe_drop_registration(license_type)
+        maybe_register_system(license_type)
```

Comment:
those two lines are a bit weird, wouldn't it be better to have one method that decides 
whether to de-register or register depending on the condition than having two methods
"`may drop registration` hold on, `may register`" ?

something like
`update_registration_state(license_type)`
and call either of the two methods inside, depending on condition

also, it would be nice to rename the methods to `drop_registration_if_x` and `register_system_if_x`, for example

Reply:
The condition to drop or register is in the function implementation to avoid having to repeat the condition here and and in another part of the code. Since the outcome for the function is not deterministic, `maybe` is the appropriat ename.

Reply:
A function to call 2 other functions and do nothing else? A bit of an overkill to me. If it were a whole process yes, but for 2 I don't see the point.

Reply:
https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/lib/cloudregister/registerutils.py#L1097

we use that pattern with no issue before